textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/utils/TestUtils.java

```diff
@@ -23,120 +16,71 @@
-            }
-        }
-        return contains;
+    /**
+     * Returns if the tupleList contains a tuple.
+     * 
+     * _ID and PAYLOAD fields are ignored in the comparison, because they are not relevant.
+     * 
+     * @param tupleList
+     * @param containsTuple
+     * @return
+     */
+    public static boolean contains(List<ITuple> tupleList, ITuple containsTuple) {
+        tupleList = Utils.removeFields(tupleList, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+        containsTuple = Utils.removeFields(containsTuple, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+        
+        return tupleList.contains(containsTuple);
     }
-
-    public static boolean containsAllResults(List<ITuple> expectedResults, List<ITuple> exactResults) {
-        expectedResults = Utils.removeFields(expectedResults, SchemaConstants.PAYLOAD);
-        exactResults = Utils.removeFields(exactResults, SchemaConstants.PAYLOAD);
-
-        if (expectedResults.size() != exactResults.size())
-            return false;
-        if (!(expectedResults.containsAll(exactResults)) || !(exactResults.containsAll(expectedResults)))
-            return false;
-
-        return true;
+    
+    /**
+     * Returns if the tupleList contains a list of tuples.
+     * 
+     * _ID and PAYLOAD fields are ignored in the comparison, because they are not relevant.
+     * 
+     * @param tupleList
+     * @param containsTupleList
+     * @return
+     */
+    public static boolean containsAll(List<ITuple> tupleList, List<ITuple> containsTupleList) {
+        tupleList = Utils.removeFields(tupleList, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+        containsTupleList = Utils.removeFields(containsTupleList, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+        
+        return tupleList.containsAll(containsTupleList);
     }
+    
+    /**
+     * Returns if the two tuple lists are equivalent (order doesn't matter)
+     * 
+     * _ID and PAYLOAD fields are ignored in the comparison, because they are not relevant.
+     * 
+     * @param expectedResults
+     * @param exactResults
+     * @return
+     */
+    public static boolean equals(List<ITuple> expectedResults, List<ITuple> exactResults) {
+        expectedResults = Utils.removeFields(expectedResults, SchemaConstants._ID, SchemaConstants.PAYLOAD);
+        exactResults = Utils.removeFields(exactResults, SchemaConstants._ID, SchemaConstants.PAYLOAD);
 
-    public static boolean containsAllResults(ArrayList<String> expectedStrings, ArrayList<String> exactStrings) {
-        if (expectedStrings.size() != exactStrings.size())
-            return false;
-        if (!(expectedStrings.containsAll(exactStrings)) || !(exactStrings.containsAll(expectedStrings)))
+        if (expectedResults.size() != exactResults.size())
             return false;
-
-        return true;
+        
+        return (expectedResults.containsAll(exactResults)) || (! exactResults.containsAll(expectedResults));

```

Comment:
I thought the right logic should be "A.containsAll(B) && B.containsAll(A)", right?